src/sdk/model/edge/edge.ts

```diff
@@ -777,8 +777,62 @@ export class Edge extends Entity {
         page_size: _pageSize
       }
     }).then((response) => {
-      const json = response.data.data as EdgeGatewayFirewallSourceObjectListJson;
-      return new EdgeGatewayFirewallSourceObjectList(json);
+      const json = response.data as EdgeGatewayFirewallObjectListJson;
+      return new EdgeGatewayFirewallObjectList(json);
+    });
+  }
+
+  /* istanbul ignore next: autogenerated */
+  async getFirewallDestinationTypes(): Promise<Array<EdgeGatewayFirewallObjectType>> {
+    return Iland.getHttp().get(`/edge-gateways/${this.uuid}/firewall/destinations`).then((response) => {
+      const json = response.data.data as Array<EdgeGatewayFirewallObjectType>;
+      return json.map((it) => new EdgeGatewayFirewallObjectType(it));
+    });
+  }
+
+  /**
+   * Get the edge firewall destination objects of the specified type.
+   * If the specified page number is larger than number of actual pages, the response will return as an empty list.
+   * @param {string | EdgeGatewayFirewallObjectType} type Type.
+   * Valid types can be retrieved from the getFirewallSourceTypes() endpoint
+   * @param {number | EdgeGatewayFirewallObjectPagingParams} pageOrPagingParams The page number or paging params.
+   * The index of the first page is 1. (Default: 1)
+   * @param {number} pageSize The page size (Optional). (Default 25)
+   * @returns {Promise<EdgeGatewayFirewallObjectList>}
+   */
+  /* istanbul ignore next: autogenerated */
+  async getFirewallDestinationObjects(type: string | EdgeGatewayFirewallObjectType,
+                                 pageOrPagingParams: number | EdgeGatewayFirewallObjectPagingParams,
+                                 pageSize?: number): Promise<EdgeGatewayFirewallObjectList> {
+    let _type = type;
+    let _page = pageOrPagingParams || 1;
+    let _pageSize = pageSize || 25;
+    if (type instanceof EdgeGatewayFirewallObjectType) {
+      _type = type.type;
+    }
+    if (pageOrPagingParams instanceof EdgeGatewayFirewallObjectPagingParams) {
+      _page = pageOrPagingParams.page;
+      _pageSize = pageOrPagingParams.pageSize;
+    }
+    return Iland.getHttp().get(`/edge-gateways/${this.uuid}/firewall/destinations/${_type}`, {
+      params: {
+        page: _page,
+        page_size: _pageSize
+      }
+    }).then((response) => {
+      const json = response.data as EdgeGatewayFirewallObjectListJson;
+      return new EdgeGatewayFirewallObjectList(json);
+    });
+  }
+
+  /**
+   * Retrieves name/ID mappings for all source and destination objects that are assigned to current firewall rules.
+   * @returns {Promise<Array<{name: string, id: string}>>}
+   */
+  /* istanbul ignore next: autogenerated */
+  async getFirewallRulesInformation(): Promise<Array<{name: string, id: string}>> {
```

Comment:
Let's give the response a formal model like we do everywhere else

Reply:
OK, let me change that